report/3.Theory/LanguageSpecification.tex

```diff
@@ -0,0 +1,92 @@
+\section{Language Specification}
+This section will describe the language designed for this project together with an explanation of which functionality and features the language will include.
+
+During the analysis face of the project it was decided that the language designed would be used as a preliminary introduction to programming.
+The language designed should therefore reflect this decision by being useful and understandable by users who are not advanced in programming and programming fundamentals.
+This decision causes both restrictions in designing the language, but also gives the freedom of designing a language for users who has no biased opinions on already established languages. 
+
+It was decided that the syntax of the language should reflect the spoken English language to some extend.
+The idea of drawing inspiration from the spoken English language is supposed to give the users of the language an easier time expressing what they want the code to do.
+In conjunction to designed after the spoken English language and the interest in utilizing high-order functions, the language design also draws inspiration from the functional programming paradigm.
+
+Since the language is supposed to be used in correlation with the Arduino platform, it is nessecary to take the Arduino's \textbf{\textit{setup}} and \textbf{\textit{loop}} functions into account.
+In order to follow this convention, a method called \textbf{\textit{loop}}, which takes a function body as a parameter, is established. 
+The code within this body is mapped to the Arduino's \textbf{\textit{loop}} method.
+Everything outside of the loop method, expressed as the "outer scope", is subsequently mapped to the Arduino's \textbf{\textit{setup}} method.
+
+Following is a rundown of the features and conventions of the languages syntax that is planned for implementation:
+
+\textbf{Types}\\
+The language include only three primitive types, namely numbers (\textbf{\textit{num}}), text-strings (\textbf{\textit{txt}}) and the boolean type (\textbf{\textit{bool}}).\\
```

Comment:
Just a question here. When I read "namely" I see that as a synonym to "mainly". So you would not write mainly followed by all the items.